client.go

```diff
@@ -24,18 +27,42 @@ func NewDefaultClient() (*Client, error) {
 
 // NewClient creates a new client, connecting to a specified port
 func NewClient(addr string) (*Client, error) {
-	conn, err := net.Dial("tcp", addr)
+	hostport := strings.SplitN(addr, ":", 2)
+	if len(hostport) == 2 {
+		return NewClientFromOptions(SetAddr(hostport[0]), SetPort(hostport[1]))
+	} else if len(hostport) == 1 {
+		return NewClientFromOptions(SetAddr("localhost"), SetPort(hostport[0]))
+	} else {
+		return NewClientFromOptions(SetAddr("localhost"), SetPort("7656"))
+	}
+}
+
+// NewClientFromOptionss creates a new client, connecting to a specified port
+func NewClientFromOptions(opts ...func(*Client) error) (*Client, error) {
+	var c Client
+	c.addr = "127.0.0.1"
+	c.port = "7656"
+	c.debug = false
+	for _, o := range opts {
+		if err := o(&c); err != nil {
+			return nil, err
+		}
+	}
+	conn, err := net.Dial("tcp", c.samaddr())
 	if err != nil {
 		return nil, err
 	}
-	if ConnDebug {
+	if c.debug {
 		conn = debug.WrapConn(conn)
 	}
-	c := &Client{
-		SamConn: conn,
-		rd:      bufio.NewReader(conn),
-	}
-	return c, c.hello()
+	c.SamConn = conn
+	c.rd = bufio.NewReader(conn)
+	return &c, c.hello()
+}
+
+//return the combined addr:port of the SAM bridge
+func (c *Client) samaddr() string {
+	return c.addr + ":" + c.port
```

Comment:
i personally practice `fmt.Sprintf("%s:%s", c.addr,c.port)` but not that big of a difference here.